sphinxcontrib/needs/directives/needtable.py

```diff
@@ -154,9 +167,13 @@ def process_needtables(app, doctree, fromdocname):
         tgroup = nodes.tgroup()
 
         # Define Table column width
-        # ToDo: Find a way to chosen to perfect width automatically.
-        for option, _title in current_needtable["columns"]:
-            if option == "TITLE":
+        colwidths = current_needtable["colwidths"]
+        for index, value in enumerate(current_needtable["columns"]):
+            option, _title = value
```

Comment:
does this work?

```suggestion
        colwidths = current_needtable["colwidths"]
        for index, option in enumerate(current_needtable["columns"].keys()):
```

Reply:
Just tested and no, as it is a list and does not know anything about ".keys()"